pyproject.toml

```diff
@@ -9,3 +9,23 @@ addopts = "--verbose --durations=0 --durations-min=0.2 --doctest-modules --mpl -
 profile = "black"
 skip_gitignore = true
 known_third_party = "pygmt"
+
+[tool.pylint.MASTER]
+# Use multiple processes to speed up Pylint. Specifying 0 will auto-detect the
+# number of processors available to use.
+jobs=0
```

Comment:
I've been getting this pylint message for a while now when I run `make lint` locally (even before this PR):

```
pylint pygmt setup.py doc/conf.py
************* Module pygmt.tests.test_blockmedian
pygmt/tests/test_blockmedian.py:30:11: E1101: Instance of 'TextFileReader' has no 'shape' member (no-member)
pygmt/tests/test_blockmedian.py:31:24: E1101: Instance of 'TextFileReader' has no 'iloc' member (no-member)
pygmt/tests/test_blockmedian.py:43:24: E1101: Instance of 'TextFileReader' has no 'iloc' member (no-member)
pygmt/tests/test_blockmedian.py:59:24: E1101: Instance of 'TextFileReader' has no 'iloc' member (no-member)
pygmt/tests/test_blockmedian.py:88:28: E1101: Instance of 'TextFileReader' has no 'iloc' member (no-member)
pygmt/tests/test_blockmedian.py:98:24: E1101: Instance of 'TextFileReader' has no 'iloc' member (no-member)
************* Module pygmt.tests.test_datasets_samples
pygmt/tests/test_datasets_samples.py:35:14: E1101: Instance of 'TextFileReader' has no 'describe' member (no-member)
pygmt/tests/test_datasets_samples.py:50:14: E1101: Instance of 'TextFileReader' has no 'describe' member (no-member)
pygmt/tests/test_datasets_samples.py:67:14: E1101: Instance of 'TextFileReader' has no 'describe' member (no-member)
pygmt/tests/test_datasets_samples.py:82:14: E1101: Instance of 'TextFileReader' has no 'describe' member (no-member)
pygmt/tests/test_datasets_samples.py:124:14: E1101: Instance of 'TextFileReader' has no 'describe' member (no-member)
pygmt/tests/test_datasets_samples.py:141:11: E1101: Instance of 'list' has no 'values' member (no-member)
************* Module pygmt.tests.test_grdtrack
pygmt/tests/test_grdtrack.py:64:24: E1101: Instance of 'TextFileReader' has no 'iloc' member (no-member)
pygmt/tests/test_grdtrack.py:94:24: E1101: Instance of 'TextFileReader' has no 'iloc' member (no-member)
************* Module pygmt.tests.test_select
pygmt/tests/test_select.py:29:11: E1101: Instance of 'TextFileReader' has no 'shape' member (no-member)
pygmt/tests/test_select.py:30:24: E1101: Instance of 'TextFileReader' has no 'median' member (no-member)
pygmt/tests/test_select.py:44:24: E1101: Instance of 'TextFileReader' has no 'median' member (no-member)
pygmt/tests/test_select.py:64:28: E1101: Instance of 'TextFileReader' has no 'median' member (no-member)
************* Module pygmt.tests.test_project
pygmt/tests/test_project.py:31:24: E1101: Instance of 'TextFileReader' has no 'iloc' member (no-member)
pygmt/tests/test_project.py:47:8: E1101: Instance of 'TextFileReader' has no 'iloc' member (no-member)
pygmt/tests/test_project.py:71:12: E1101: Instance of 'TextFileReader' has no 'iloc' member (no-member)
************* Module pygmt.tests.test_blockm
pygmt/tests/test_blockm.py:32:11: E1101: Instance of 'TextFileReader' has no 'shape' member (no-member)
pygmt/tests/test_blockm.py:33:24: E1101: Instance of 'TextFileReader' has no 'iloc' member (no-member)
pygmt/tests/test_blockm.py:46:24: E1101: Instance of 'TextFileReader' has no 'iloc' member (no-member)
pygmt/tests/test_blockm.py:62:24: E1101: Instance of 'TextFileReader' has no 'iloc' member (no-member)
pygmt/tests/test_blockm.py:91:28: E1101: Instance of 'TextFileReader' has no 'iloc' member (no-member)
pygmt/tests/test_blockm.py:101:24: E1101: Instance of 'TextFileReader' has no 'iloc' member (no-member)
pygmt/tests/test_blockm.py:111:11: E1101: Instance of 'TextFileReader' has no 'shape' member (no-member)
pygmt/tests/test_blockm.py:112:24: E1101: Instance of 'TextFileReader' has no 'iloc' member (no-member)

------------------------------------------------------------------
Your code has been rated at 9.74/10 (previous run: 9.74/10, +0.00)

make: *** [Makefile:58: lint] Error 2
```


Based on https://github.com/PyCQA/pylint/issues/4577#issuecomment-1000245962, the solution would be to do this:

```suggestion
jobs=0
# As a temporary workaround for https://github.com/PyCQA/pylint/issues/4577
init-hook = "import astroid; astroid.context.InferenceContext.max_inferred = 500"
```
but I'm not sure if we should add it, since the GitHub Actions style checks pass anyway. Does `make lint` run fine without errors for you?

Reply:
Yes, I'm getting the same errors locally for quite a long time.

Reply:
Ok, check if adding that line works for you, and maybe we can add it in here :smile:

Reply:
It doesn't work for me. I'm having pylint 2.12.2 and astroid 2.9.3

Reply:
Hmm, I've got pylint 2.12.2 and astroid 2.9.0. Let me see if upgrading astroid to 2.9.3 breaks it.

Reply:
`generated-members=pandas.*` works, but I still have one error:
```
************* Module pygmt.tests.test_datasets_samples
pygmt/tests/test_datasets_samples.py:141:11: E1101: Instance of 'list' has no 'values' member (no-member)
```

Reply:
Should be possible to refactor that test to not use `.values` I think.